config/routes.rb

```diff
@@ -49,12 +47,22 @@
     post "/webhooks/", to: "stripe#webhooks", as: "stripe_webhooks"
   end
 
+  # Mount development tools
   if Rails.env.development?
     namespace :development do
       resources :design, only: [:index]
       resources :embed_mock, only: [:index]
     end
   end
 
+  # Mount the sidekiq dashboard, secured with Basic Auth
+  require "sidekiq/web"
+
+  unless Rails.env.development?
+    Sidekiq::Web.use Rack::Auth::Basic do |username, password|
+      username == ENV["SIDEKIQ_USERNAME"] &&
+        password == ENV["SIDEKIQ_PASSWORD"]
+    end
+  end
   mount Sidekiq::Web => "/sidekiq"
```

Comment:
Another idea would have been to only mount sidekiq in development, but this works too I guess